starboard/nplb/compiler_compliance/cpp20_support.cc

```diff
@@ -0,0 +1,75 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the aLicense is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <iostream>
+#include <memory>
+#include <numeric>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+namespace starboard {
+namespace nplb {
+namespace compiler_compliance {
+namespace {
+
+// These examples are taken after referring to
+// 1) The chromium C++20 allowlist:
+// https://chromium.googlesource.com/chromium/src/+/refs/branch-heads/5735/styleguide/c++/c++-features.md
+// 2) cpp reference : https://en.cppreference.com/w/cpp
+
+// Test std::string ends_with support
+void test_string_ends_with() {
+  bool result = std::string("foobar").ends_with("bar");
+}
+
+#if __cpp_lib_assume_aligned >= 201811L
+// Test std::assume_aligned support
+void test_assume_aligned(int* p) {
+  int* p1 = std::assume_aligned<256>(p);
+}
+#else
+// Fallback implementation if std::assume_aligned is not supported
```

Comment:
Are you adding some fallback implementation here ? If not, i'd rather actually remove `std::assume_aligned` test here entirely, with the expectation that it's not allowed in our code for now.

Reply:
+1 
Otherwise we would have to check `__cpp_lib_assume_aligned` everywhere we try to use the API.

Reply:
gotcha , will remove it.

we'll need to add it back at some point( as it's allowlisted in chrome trunk ) but i can remove it for now.